plugin/src/main/groovy/com/novoda/staticanalysis/internal/ktlint/KtlintConfigurator.groovy

```diff
@@ -46,29 +44,34 @@ class KtlintConfigurator implements Configurator {
             configureKtlintExtension(config)
 
             project.afterEvaluate {
-
-                project.plugins.withId("kotlin") {
-                    configureKotlinProject()
-                }
-                project.plugins.withId("kotlin2js") {
-                    configureKotlinProject()
-                }
-                project.plugins.withId("kotlin-platform-common") {
-                    configureKotlinProject()
-                }
-                project.plugins.withId("org.jetbrains.kotlin.multiplatform") {
-                    configureKotlinProject()
-                }
-                project.plugins.withId('com.android.application') {
-                    configureAndroidWithVariants(variantFilter.filteredApplicationVariants)
-                }
-                project.plugins.withId('com.android.library') {
-                    configureAndroidWithVariants(variantFilter.filteredLibraryVariants)
+                createTask(project, 'triggerKtlint', ToolTriggerTask) { trigger ->
```

Comment:
Our `ktlint` integration is the most special. We dynamically look at the live collection of tasks and pick up ktlint tasks and configure them later. And we create `collectViolations` tasks based on availability to `ktlint` tasks. When we want to keep all of these lazy, it is simply a deadlock.

The only solution that I can come up with is to introduce another task called `trigger`. Calling this will resolve all ktlint tasks with no problem. When this task is called, we are sure that it is safe to configure everything.